lib/logstash/inputs/http_poller.rb

```diff
@@ -91,77 +73,101 @@ def run(queue)
     end
   end
 
+  private
   def run_once(queue)
     @urls.each do |name, url|
-      @logger.debug? && @logger.debug("Will get url '#{name}' '#{url}")
-      client.async.get(url).
-        on_success {|response| handle_success(queue, name, url, response)}.
-        on_failure {|exception| handle_failure(queue, name, url, exception)}
+      request_async(queue, name, url)
+    end
+
+    # Some exceptions are only returned here! There is no callback,
+    # for example, if there is a bad port number.
+    client.execute!.each_with_index do |resp, i|
```

Comment:
Yeah, I need to dig into this. I tried to unify this earlier today and hit some inconsistencies. This may need to be passed back upstream to Manticore as a bug. It seems that the failure handler misses some exceptions.